pkg/model/model.go

```diff
@@ -107,6 +107,17 @@ type Model interface {
 	CreateMetadataConfiguration(ctx context.Context, metadata *MetadataConfiguration) error
 	GetMetadataConfiguration(ctx context.Context, repoDID string) (*MetadataConfiguration, error)
 	DeleteMetadataConfiguration(ctx context.Context, repoDID string) error
+
+	CreateModerationDelegation(ctx context.Context, delegation *ModerationDelegation) error
+	DeleteModerationDelegation(ctx context.Context, rkey string) error
+	GetModerationDelegation(ctx context.Context, streamerDID, moderatorDID string) (*ModerationDelegation, error)
+	GetModerationDelegations(ctx context.Context, streamerDID, moderatorDID string) ([]*ModerationDelegation, error)
+	GetModeratorDelegations(ctx context.Context, moderatorDID string) ([]*ModerationDelegation, error)
+	GetStreamerModerators(ctx context.Context, streamerDID string) ([]*ModerationDelegation, error)
```

Comment:
So most of Streamplace doesn't do this right now but I'm trying as much as possible to push the model layer to only ingest and return atproto structs from the `streamplace` package so we're not passing around `model` stuff elsewhere. That way the "model" indexing layer stays mostly self-contained. See the `BroadcastOrigin` stuff for a good example of how to do it

This might require to create a new non-record lexicon for "moderation delegation view" or something that wraps the record with an "author" but that'll be useful in other contexts too I think